Cython/Compiler/Pythran.py

```diff
@@ -76,7 +76,8 @@ def index_code(idx):
         raise ValueError("unsupported indexing type %s!" % idx.type)
 
     indexing = ",".join(index_code(idx) for idx in indices)
-    return type_remove_ref("decltype(std::declval<%s>()[%s])" % (pythran_type(type_), indexing))
+    index = '[%s]' if len(indices) == 1 else '(%s)'
+    return type_remove_ref(("decltype(std::declval<%s>()" + index) % (pythran_type(type_), indexing))
```

Comment:
Ah, there was a closing parenthesis missing here, after the `index`. With that, it's now compiling and giving the correct result for me.